scyllacloud/hostdialer.go

```diff
@@ -96,8 +96,12 @@ func (s *SniHostDialer) dialInitialContactPoint(ctx context.Context, host *gocql
 		return nil, fmt.Errorf("can't get root CA from configuration: %w", err)
 	}
 
-	return s.connect(ctx, s.dialer, host.HostnameAndPort(), &tls.Config{
-		ServerName:         host.Hostname(),
+	contextConf := s.connConfig.Contexts[s.connConfig.CurrentContext]
```

Comment:
what if the key is missing?

Reply:
can the map be nil?

Reply:
added error handling, nil map is also handled

Reply:
would be nice to cover it with a test

Reply:
added tests